lxc/storage_volume.go

```diff
@@ -1103,6 +1104,15 @@ func (c *cmdStorageVolumeEdit) run(cmd *cobra.Command, args []string) error {
 		isSnapshot = true
 	}
 
+	// Get editable fields for snapshots
+	var snapshotEditableFields []string
+	var snapshotEditableFieldsMsg string
+	if isSnapshot {
+		// Use a temporary instance to determine editable fields
+		tempSnapshot := api.StorageVolumeSnapshot{}
+		snapshotEditableFields = c.getEditableYAMLFields(tempSnapshot.Writable())
```

Comment:
to simplify tracking where these variables are defined and used, how about creating a function inside this function like so:

```go
snapshotEditableFieldsMsg := func(v api.StorageVolumeSnapshotPut) []string{
     return fmt.Sprintf(i18n.G("Only %s field(s) can be modified for snapshot volumes"), c.formatFieldList(c.getEditableYAMLFields(v)))
}
```